### DISCUSSION TITLE
[@types/yargs] argv's type is missing camelCased variants of hyphenated options

### DISCUSSION THREAD

**Discussion by kevinji** [DefinitelyTyped/DefinitelyTyped#57069] [Issues with a @types package]
yargs automatically generates camelCased versions of hyphenated options (e.g. `full-page` -> `fullPage`) inside of the `argv` object, but the TypeScript type for yargs doesn't include the camelCased variants. This seems like an omission, but I'm not familiar enough with the TypeScript type system to know whether this is possible to fix or not.

Issue exists in the yargs repo as yargs/yargs#2055.

[a comment by an automated account is not shown]

**Comment by forivall** [comment]
Hmm, seems that i worked on this in January when typescript released 4.1 which supports template string typings, but i didn't end up finishing it. I forget if there was anything missing: https://github.com/DefinitelyTyped/DefinitelyTyped/pull/50722

**Comment by bcoe** [comment]
@forivall this would be an awesome feature to land 😄 (and I think is probably well beyond my TypeScript skill).

Perhaps I can look at your PR over a coffee on a weekend though, and try to help get it over the finish line (unless you beat me to the punch).
